core/server/api_container/server/startosis_engine/kurtosis_instruction/add_service/add_service.go

```diff
@@ -172,6 +172,8 @@ func (builtin *AddServiceCapabilities) Interpret(locatorOfModuleInWhichThisBuilt
 	if err != nil {
 		return nil, startosis_errors.WrapWithInterpretationError(err, "An error occurred while persisting return value for service '%v'", serviceName)
 	}
+	builtin.interpretationTimeValueStore.PutServiceConfig(builtin.serviceName, builtin.serviceConfig)
```

Comment:
hmm i would think serialization could have failures

Reply:
There's no serialization of `serviceConfig`s happening, the `ServiceConfig` objects are stored in memory.